harness/determined/util.py

```diff
@@ -325,3 +328,28 @@ def match_legacy_trial_class(arg: str) -> bool:
 
 def legacy_trial_entrypoint_to_script(trial_entrypoint: str) -> List[str]:
     return ["python3", "-m", "determined.exec.harness", trial_entrypoint]
+
+
+@contextlib.contextmanager
+def forward_signals(p: subprocess.Popen, *signums: signal.Signals) -> Iterator[None]:
+    """Forward a list of signals to a subprocess, restoring the original handlers afterwards."""
+    if not signums:
```

Comment:
I guess the conditionality here could be useful in the future? For now I notice that we always call it with one argument (i.e. without specific signals).

Reply:
:shrug: I could go either way.